src/CRUD.php

```diff
@@ -152,13 +152,13 @@ public function setModel(Model $m, $fields = null): Model
     protected function initActionExecutor(Generic $action)
     {
         $executor = $this->getExecutor($action);
-        $executor->onHook('afterExecute', function ($ex, $return, $id) use ($action) {
+        $executor->onHook(ActionExecutor\Basic::HOOK_AFTER_EXECUTE, function ($ex, $return, $id) use ($action) {
             return $this->jsExecute($return, $action);
         });
 
         if ($executor instanceof UserAction) {
             foreach ($this->onActions as $k => $onAction) {
-                $executor->onHook('onStep', function ($ex, $step, $form) use ($onAction, $action) {
+                $executor->onHook(UserAction::HOOK_STEP, function ($ex, $step, $form) use ($onAction, $action) {
```

Comment:
And here is need to know that hook step hook belongs to UserAction class and previous hook after execute hook belongs to ActionExecutor\Basic class... No, I do not like that.
Probably here it should better be `$executor->onHook($executor::HOOK_STEP, ...)` and I do not care in which of executor parent classes this constant is defined.

Same syntax probably could be used everywhere `$obj->onHook($obj::HOOK_NAME, ...)` instead of using exact class names.